File: app/src/main/java/com/hellmund/primetime/main/DiscoverMoreFragment.java
package com.hellmund.primetime.main;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.hellmund.primetime.R;

import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

public class DiscoverMoreFragment extends Fragment {

    private OnInteractionListener mCallback;
    private Unbinder mUnbinder;

    public static DiscoverMoreFragment newInstance() {
        return new DiscoverMoreFragment();
    }

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_discover_more, container, false);
        mUnbinder = ButterKnife.bind(this, view);
        // mCallback = (OnInteractionListener) getContext();
        return view;
    }

    @OnClick(R.id.discover_more_btn)
    public void discoverMore() {
        mCallback.openCategories();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mUnbinder.unbind();
    }

    public interface OnInteractionListener {
        void openCategories();
    }

}

File: app/src/main/java/com/hellmund/primetime/ui/main/DiscoverMoreFragment.kt
package com.hellmund.primetime.ui.main

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.hellmund.primetime.R
import kotlinx.android.synthetic.main.fragment_discover_more.*

class DiscoverMoreFragment : Fragment() {

    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? = inflater.inflate(R.layout.fragment_discover_more, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        discoverMoreButton.setOnClickListener {
            val activity = requireActivity() as MainActivity
            activity.openSearch()
        }
    }

    companion object {
        fun newInstance(): DiscoverMoreFragment = DiscoverMoreFragment()
    }

}
